src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingStackTraceService.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.StackTraceExplorer;
+using System.Collections.Immutable;
+using System;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.API
+{
+    internal sealed class UnitTestingStackTraceService
+    {
+        private readonly IStackTraceExplorerService _stackTraceService;
+
+        public UnitTestingStackTraceService(HostWorkspaceServices services)
```

Comment:
Can this be changed to accept Workspace instead? This can then be used to access Workspace.Services here. The Workspace can also be threaded down to Frame and Definition below to avoid the need for caller to pass it when calling Definition.TryNavigateToAsync below.

Reply:
Sure, this was based off another item. Whatever is easiest we can accomodate